### ISSUE TITLE
Negative Kron radii in SourceCatalog

### ISSUE THREAD

**Issue by drtobybrown** [astropy/photutils#1122]
I'm probably missing something but I wasn't expecting to get negative kron radii in the SourceCatalog..

Is this a bug or quirk of how kron radii are calculated? If the latter, why is it occurring and how do I avoid this in future?

Fits r-band image used is [here](https://www.dropbox.com/s/w9pzkj8iqqjuq5p/NGC4450_SDSS_r.fits?dl=0) and steps to reproduce are:

```
data, header = fits.getdata(file, header=True)

# define a 2D detection threshold    
bkg_estimator = MedianBackground()
boxsize = int(data.shape[0] / 5)
sigma_clip = SigmaClip(sigma=3.)

bkg = Background2D(data, boxsize, filter_size=50, sigma_clip=sigma_clip,
                   bkg_estimator=bkg_estimator)
threshold = bkg.background + (3. * bkg.background_rms)

# build segmentation map    
sigma = 3.0 * gaussian_fwhm_to_sigma  # FWHM = 3.
npixels = 50
kernel = Gaussian2DKernel(sigma, x_size=3, y_size=3)
kernel.normalize()
segm = detect_sources(data, threshold, npixels=npixels, filter_kernel=kernel)

# measure the properties of the detected sources defined in the segmentation image 
kron_params = (['correct', 1, 0.0, 'exact', 5])
cat = source_properties((data-bkg.background) * u.Jy/u.pix
                        , segm
                        , background=bkg.background * u.Jy/u.pix
                        , wcs=wcs
                        , kron_params=kron_params)

for obj in cat:
    if obj.kron_radius < 0:
        print(obj)
        print(obj.kron_radius)
```

**Comment by drtobybrown** [comment]
As a follow up negative radii also cause problems for .to_table() method

**Comment by larrybradley** [comment]
@drtobybrown Many thanks for reporting this, and providing the data and minimal example to help me debug!  The issue is there are many negative pixels surrounding this particular source (perhaps due to oversubtraction of the local background?), and that has triggered a bug in the elliptical radial-moment calculation:

Here's the source with negative Kron radius.  The yellow pixels in the right panel are all negative and the ellipse contains the pixels used to generate the radial moments:
![download](https://user-images.githubusercontent.com/4992897/99571987-04fc0c80-29a2-11eb-8451-24e1d2c91b5b.png)

I'll have a fix for this next week.  It looks like the `sep` package flags such sources and simply returns `kron_radius=0`.  A better approach may be to exclude the negative pixels and then handle the special case where all pixels are negative.  In any case, I'll have a fix soon.

**Comment by drtobybrown** [comment]
That makes sense. For what it's worth, I agree about the better approach being to exclude negative pix. It may also be useful to give a warning to the user letting them know the bg is potentially over-subtracted.

**Comment by larrybradley** [comment]
@drtobybrown I fixed the negative Kron radius issue in #1132.  I decided to take the approach that if the numerator or denominator in the Kron radius formula (k_r = Sum(r_i * I_i) / Sum(I_i)) that the Kron radius should be undefined (np.nan). In that case, a CircularAperture with radius `kron_radius[2]` will be used to compute the Kron flux.  If `kron_radius[2] <= 0`, then the `kron_flux` will be nan.
